iceoryx_posh/test/moduletests/test_popo_toml_gateway_config_parser.cpp

```diff
@@ -20,201 +20,268 @@
 using namespace ::testing;
 using ::testing::_;
 
-// ======================================== Helpers ======================================== //
+using namespace iox::config;
 
+// ======================================== Helpers ======================================== //
+const std::string TestFile = "gwconfig_test.tmp";
+#ifndef _WIN32
+const std::string TempPath = "/tmp";
+const std::string TestFilePath = TempPath + "/" + TestFile;
+#else
+const std::string TempPath = std::getenv("TEMP");
+const std::string TestFilePath = TempPath + "\\" + TestFile;
+#endif
 
 // ======================================== Fixture ======================================== //
-class TomlGatewayConfigParserTest : public Test
+class TomlGatewayConfigParser_Test : public TestWithParam<std::tuple<std::string, bool>>
```

Comment:
You could define your own type for `std::tuple<std::string, bool>` and just use the custom type in `::testing::Values` as well. Check out current master of this file!